prusti-encoder/src/encoders/mir_impure.rs

```diff
@@ -161,29 +161,49 @@ impl TaskEncoder for MirImpureEncoder {
             posts.push(local_defs.locals[mir::RETURN_PLACE].impure_pred);
             posts.extend(spec_posts);
 
-            let mut visitor = EncoderVisitor {
-                vcx,
-                deps,
-                local_decls: &body.local_decls,
-                //ssa_analysis,
-                fpcs_analysis,
-                local_defs,
 
-                tmp_ctr: 0,
+             // TODO: dedup with mir_pure
+             let attrs = vcx.tcx.get_attrs_unchecked(def_id);
+             let is_trusted = attrs
+                 .iter()
+                 .filter(|attr| !attr.is_doc_comment())
+                 .map(|attr| attr.get_normal_item())
+                 .any(|item| {
+                     item.path.segments.len() == 2
+                         && item.path.segments[0].ident.as_str() == "prusti"
+                         && item.path.segments[1].ident.as_str() == "trusted"
```

Comment:
There is an established system for checking if a function is trusted, please do not circumvent it. It's the same thing you used to find if a function is `pure` here: https://github.com/tillarnold/prusti-dev/blob/fa4aa23074ba6712abfe8bc7f0715fc509be5bc0/prusti-encoder/src/lib.rs#L145-L156
This is bad because if we were to change the text of the "trusted" tag then this would break.